src/usePermissions.js

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@tanstack/react-query';
+import { abi, address } from '@vderunov/whitelist-contract/deployments/11155420/Whitelist';
+import { Contract } from 'ethers';
+import { useSynthetix } from './useSynthetix';
+
+function usePermissions() {
+  const [synthetix] = useSynthetix();
+  const contract = new Contract(address, abi, synthetix.provider);
```

Comment:
should be inside async function. Otherwise it will be created every time hook is ever rendered in any component instead of only once when actual call is executed.

Reply:
fixed